Assync

A1tehr opened the pull request:
Добавлены асинхронные методы подключения

A1tehr committed:
Init branch and alpha version

- Добавлены асинхронные методы
- Добавлена защита от повторных подключений к БД.

A1tehr committed:
Test

A1tehr committed:
Bug

A1tehr committed:
Delete probel

anst-foto committed:
Update

A1tehr committed:
Alpha

SergeySamodurovS commented on DBconnect/DBconnection.cs:
```diff
@@ -148,14 +279,53 @@ public int UpdateQuery(string sql)
-            connection.Close();
+            if (connection.Ping())
+            {
+                connection.Close();
```
Можно после закрытия подключения к БД сообщить об этом с помощью делегата. Так же и в асинхронном методе.

SergeySamodurovS commented on DBconnect/DBconnection.cs:
```diff
@@ -78,6 +143,25 @@ public void ConnectDB()
+        {
+            if (connection != null && connection.Ping())
+            {
+                error?.Invoke("Подключение уже осуществлено [async connection]");
```
```suggestion
                error?.Invoke("Подключение уже осуществлено [async connection]");
```
Не факт, что подключение ранее было установлено асинхронно.

SergeySamodurovS replied:
Не факт, что подключение ранее было установлено асинхронно.

anst-foto reviewed (changes requested):
Переделать конструктор с параметром типа bool. Отказаться от конструктора с одним аргументом и исключить из него методы подключения к БД

A1tehr committed:
Relis

- Убраны лишние конструкторы
- Удалены методы подключение к БД из конструкторов.
- Ищменены аргументы в эвенты
